producer_client/spec/lib/animals/resources/cat_spec.rb

```diff
@@ -28,4 +28,38 @@
 
     it_behaves_like 'listing cats', [garfield, felix, sylvester]
   end
+
+  describe '#create' do
+    name = 'Garfield'
+    age = 39
+
+    before do
+      request = {
+        cat: {
+          name: name,
+          age: age,
+        }
+      }
+      response = {
+        data: {
+          id: Pact.like('1'),
+          type: 'cat',
+          attributes: {
+            name: name,
+            age: age,
+          },
+        }
+      }
+      headers = {
+        'Content-Type' => 'application/json',
+      }
+
+      animals_service
+        .upon_receiving('a request to create a cat')
+        .with(method: :post, path: '/cats', body: request, headers: headers)
+        .will_respond_with(status: 201, body: response)
+    end
+
+    it_behaves_like 'creating a cat', name, age
```

Comment:
These shared examples really have helped ensure that the behavior of the real and fake resources are the same.